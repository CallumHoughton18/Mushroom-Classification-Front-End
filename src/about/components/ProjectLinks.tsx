import React, {FunctionComponent} from "react";
import GithubLogo from "../../shared/components/Logos/GithubLogo";
import DockerLogo from "../../shared/components/Logos/DockerLogo";

type ProjectLinksType = {
    PrimaryLogoColor: string;
    SecondaryLogoColor: string;
    style: string;
};

const ProjectLinks: FunctionComponent<ProjectLinksType> = (props) => {
    return (
        <div className={props.style}>
            <GithubLogo
                LogoPrimaryColor={props.PrimaryLogoColor}
                Link="https://github.com/CallumHoughton18/Mushroom-Classification-Front-End"
            />
            <DockerLogo
                LogoPrimaryColor={props.PrimaryLogoColor}
                LogoSecondaryColor={props.SecondaryLogoColor}
                Link="https://hub.docker.com/repository/docker/callumhoughton22/mushroom-classification-front-end"
            />
        </div>
    );
};

export default ProjectLinks;
