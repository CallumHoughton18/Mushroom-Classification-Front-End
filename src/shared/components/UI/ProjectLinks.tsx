import React, {FunctionComponent} from "react";
import GithubLogo from "../Logos/GithubLogo";
import DockerLogo from "../Logos/DockerLogo";

type ProjectLinksType = {
    PrimaryLogoColor: string;
    SecondaryLogoColor: string;
    style: string;
};

const ProjectLinks: FunctionComponent<ProjectLinksType> = (props) => {
    return (
        <div className={props.style}>
            <GithubLogo LogoPrimaryColor={props.PrimaryLogoColor} />
            <DockerLogo
                LogoPrimaryColor={props.PrimaryLogoColor}
                LogoSecondaryColor={props.SecondaryLogoColor}
            />
        </div>
    );
};

export default ProjectLinks;
