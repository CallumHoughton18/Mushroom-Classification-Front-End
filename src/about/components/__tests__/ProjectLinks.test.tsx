import React from "react";
import {render} from "@testing-library/react";

import ProjectLinks from "../ProjectLinks";

describe("<ProjectLinks /> render test", () => {
    it("should display Github and Docker project links", () => {
        const renderedLinks = render(
            <ProjectLinks PrimaryLogoColor="green" SecondaryLogoColor="blue" style="" />
        );
        expect(renderedLinks).toMatchSnapshot();
    });
});
