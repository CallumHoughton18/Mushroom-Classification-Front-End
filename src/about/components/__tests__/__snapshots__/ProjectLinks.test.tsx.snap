// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectLinks /> render test should display Github and Docker project links 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class=""
      >
        <a
          href="https://github.com/CallumHoughton18/Mushroom-Classification-Front-End"
        >
          <svg
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
          >
            <path
              d="m12 .5c-6.63 0-12 5.28-12 11.792 0 5.211 3.438 9.63 8.205 11.188.6.111.82-.254.82-.567 0-.28-.01-1.022-.015-2.005-3.338.711-4.042-1.582-4.042-1.582-.546-1.361-1.335-1.725-1.335-1.725-1.087-.731.084-.716.084-.716 1.205.082 1.838 1.215 1.838 1.215 1.07 1.803 2.809 1.282 3.495.981.108-.763.417-1.282.76-1.577-2.665-.295-5.466-1.309-5.466-5.827 0-1.287.465-2.339 1.235-3.164-.135-.298-.54-1.497.105-3.121 0 0 1.005-.316 3.3 1.209.96-.262 1.98-.392 3-.398 1.02.006 2.04.136 3 .398 2.28-1.525 3.285-1.209 3.285-1.209.645 1.624.24 2.823.12 3.121.765.825 1.23 1.877 1.23 3.164 0 4.53-2.805 5.527-5.475 5.817.42.354.81 1.077.81 2.182 0 1.578-.015 2.846-.015 3.229 0 .309.21.678.825.56 4.801-1.548 8.236-5.97 8.236-11.173 0-6.512-5.373-11.792-12-11.792z"
              fill="green"
            />
          </svg>
        </a>
        <a
          href=""
        >
          <svg
            height="100%"
            version="1.1"
            viewBox="0 0 512 512"
            width="100%"
          >
            <path
              d="M512,256c0,141.385-114.615,256-256,256S0,397.385,0,256S114.615,0,256,0
            c130.497,0,238.184,97.646,253.994,223.859c1.149,9.122,1.818,18.39,1.964,27.784C511.99,253.095,512,254.548,512,256z"
              fill="green"
            />
            <path
              d="M511.958,251.643c-21.859,26.624-64.69,24.545-64.69,24.545
                c-26.415,113.8-185.94,171.719-296.699,143.266C39.821,391.011,55.056,258.915,55.056,258.915h324.138
                c26.749,0,28.787-8.129,28.787-8.129c-23.709-56.226,12.873-86.371,12.873-86.371c28.453,5.089,35.558,53.854,35.558,53.854
                c23.353-4.19,42.109,0.794,53.582,5.59C511.143,232.981,511.812,242.249,511.958,251.643z"
              fill="blue"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="89.6"
              y="207.433"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="146.641"
              y="207.433"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="317.743"
              y="207.433"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="260.712"
              y="207.433"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="260.712"
              y="101.763"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="203.671"
              y="207.433"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="146.641"
              y="154.593"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="260.712"
              y="154.593"
            />
            <rect
              fill="blue"
              height="41.32"
              width="45.39"
              x="203.671"
              y="154.593"
            />
          </svg>
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class=""
    >
      <a
        href="https://github.com/CallumHoughton18/Mushroom-Classification-Front-End"
      >
        <svg
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
        >
          <path
            d="m12 .5c-6.63 0-12 5.28-12 11.792 0 5.211 3.438 9.63 8.205 11.188.6.111.82-.254.82-.567 0-.28-.01-1.022-.015-2.005-3.338.711-4.042-1.582-4.042-1.582-.546-1.361-1.335-1.725-1.335-1.725-1.087-.731.084-.716.084-.716 1.205.082 1.838 1.215 1.838 1.215 1.07 1.803 2.809 1.282 3.495.981.108-.763.417-1.282.76-1.577-2.665-.295-5.466-1.309-5.466-5.827 0-1.287.465-2.339 1.235-3.164-.135-.298-.54-1.497.105-3.121 0 0 1.005-.316 3.3 1.209.96-.262 1.98-.392 3-.398 1.02.006 2.04.136 3 .398 2.28-1.525 3.285-1.209 3.285-1.209.645 1.624.24 2.823.12 3.121.765.825 1.23 1.877 1.23 3.164 0 4.53-2.805 5.527-5.475 5.817.42.354.81 1.077.81 2.182 0 1.578-.015 2.846-.015 3.229 0 .309.21.678.825.56 4.801-1.548 8.236-5.97 8.236-11.173 0-6.512-5.373-11.792-12-11.792z"
            fill="green"
          />
        </svg>
      </a>
      <a
        href=""
      >
        <svg
          height="100%"
          version="1.1"
          viewBox="0 0 512 512"
          width="100%"
        >
          <path
            d="M512,256c0,141.385-114.615,256-256,256S0,397.385,0,256S114.615,0,256,0
            c130.497,0,238.184,97.646,253.994,223.859c1.149,9.122,1.818,18.39,1.964,27.784C511.99,253.095,512,254.548,512,256z"
            fill="green"
          />
          <path
            d="M511.958,251.643c-21.859,26.624-64.69,24.545-64.69,24.545
                c-26.415,113.8-185.94,171.719-296.699,143.266C39.821,391.011,55.056,258.915,55.056,258.915h324.138
                c26.749,0,28.787-8.129,28.787-8.129c-23.709-56.226,12.873-86.371,12.873-86.371c28.453,5.089,35.558,53.854,35.558,53.854
                c23.353-4.19,42.109,0.794,53.582,5.59C511.143,232.981,511.812,242.249,511.958,251.643z"
            fill="blue"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="89.6"
            y="207.433"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="146.641"
            y="207.433"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="317.743"
            y="207.433"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="260.712"
            y="207.433"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="260.712"
            y="101.763"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="203.671"
            y="207.433"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="146.641"
            y="154.593"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="260.712"
            y="154.593"
          />
          <rect
            fill="blue"
            height="41.32"
            width="45.39"
            x="203.671"
            y="154.593"
          />
        </svg>
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
